Workflow ci

parteekcoder opened the pull request:
fixes issue https://github.com/llaske/sugarizer-server/issues/293

@llaske @NikhilM98 I tested this workflow on my repository all test are passing, you can review this now

Commit:
Merge branch 'dev'

parteekcoder committed:
Create BuildAndTest.yml

cperreault committed:
Fix container image name

Commit:
Merge pull request #374 from cperreault/docs-migrate

Refer to the MongoDB container in the migrate to v1.5.0 docs

parteekcoder committed:
Update BuildAndTest.yml

parteekcoder committed:
Update BuildAndTest.yml

parteekcoder committed:
Update BuildAndTest.yml

parteekcoder committed:
Create main.yml

parteekcoder committed:
Update BuildAndTest.yml

parteekcoder commented:
@NikhilM98 would you like to assign me some task to work upon

llaske commented on .github/workflow/BuildAndTest.yml:
```diff
@@ -0,0 +1,33 @@
+name: CI
+on:
+  push:
+    branches: [dev,master]
```
I suggest to limit the action to dev branch because no PR is accepted on the master branch

llaske commented on .github/workflow/BuildAndTest.yml:
```diff
@@ -0,0 +1,33 @@
+  push:
+    branches: [dev,master]
+  pull_request:
+    branches: [dev,master]
```
I suggest to limit the action to dev branch because no PR is accepted on the master branch

parteekcoder replied:
that's is right , but it will just increase checks

NikhilM98 commented:
@parteekcoder the PR that addresses the Eslint issues has been merged. The workflow can now be updated to include the Eslint checks.

parteekcoder committed:
Merge branch 'dev' into workflow_CI

parteekcoder committed:
adding lint

parteekcoder commented:
@NikhilM98 I think `lint` script in package.json also have to be updated

parteekcoder committed:
Merge remote-tracking branch 'upstream/dev' into workflow_CI

NikhilM98 commented:
@parteekcoder you need to run ‘npm ci’ before running ‘npm run lint’

parteekcoder committed:
Update BuildAndTest.yml

parteekcoder commented:
hey @NikhilM98 give me some time , have to make some changes

parteekcoder committed:
fixes eslint issue

NikhilM98 commented:
Make sure to remove all the unnecessary changes and test how the pipeline works if an ESLint check fails. Also, make capitalization consistent.

parteekcoder commented:
> Make sure to remove all the unnecessary changes and test how the pipeline works if an ESLint check fails. Also, make capitalization consistent.

sure

parteekcoder commented:
@llaske @NikhilM98 this is my report

passing test:
https://github.com/parteekcoder/sugarizer-server/pull/6
failing test:
https://github.com/parteekcoder/sugarizer-server/pull/5    i added incorrect indentation to fail this test
![image](https://user-images.githubusercontent.com/100110395/222956036-e8de76f9-6983-4523-9bca-eb1abbb71f6e.png)

parteekcoder committed:
Update migrate.md

parteekcoder committed:
Update migrate.md

required changes

Update testingWorkflow.yml

Update testingWorkflow.yml

Update .eslintignore

Update .eslintignore

NikhilM98 commented on .eslintignore:
```diff
@@ -22,3 +22,5 @@ api/controller/utils/qrCodeUtil.js
 
 build/
+
+sugarizer/
```
Can you explain why you’ve added this? It’s probably not needed.

NikhilM98 replied:
@parteekcoder this issue still needs to be addressed.

NikhilM98 commented on .github/workflows/BuildAndTest.yml:
```diff
@@ -0,0 +1,35 @@
+        run: npm ci
+      - name: Running Eslint
+        run: npm run lint
+      - name: Unit testing
```
Replace ‘testing’ with ‘Testing’.

NikhilM98 replied:
And maybe use a more standard name for the YAML file

parteekcoder committed:
required changes

parteekcoder commented:
@NikhilM98 plz tell me if you would like to change

NikhilM98 commented:
@parteekcoder there are unresolved issues with this PR. Can you go through the comments left by me and Lionel and update the PR?

parteekcoder commented:
@NikhilM98 as I resolved one conversation
and for the other I said it necessary change in .eslintignore file which is to be made the reason I provided in the conversation I resolved

NikhilM98 commented:
> and for the other I said it necessary change in .eslintignore file which is to be made the reason I provided in the conversation I resolved

The changes you have made in `.eslintignore` should not be necessary. Can you show a case in which the workflow fails if you have not made the changes you did in the `.eslintignore` file?

NikhilM98 commented on .github/workflows/testingWorkflow.yml:
```diff
@@ -0,0 +1,35 @@
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [15.x]
+        mongodb-version: ['6.0']
```
Why is one version in single quotes while the other one is not? Does it works if you use `6.0` instead of `'6.0'`?

parteekcoder committed:
Update testingWorkflow.yml

parteekcoder commented:
> > and for the other I said it necessary change in .eslintignore file which is to be made the reason I provided in the conversation I resolved
> 
> The changes you have made in `.eslintignore` should not be necessary. Can you show a case in which the workflow fails if you have not made the changes you did in the `.eslintignore` file?

@NikhilM98 please see  this https://github.com/parteekcoder/sugarizer-server/actions/runs/4342453548/jobs/7583289146

Then it also checks sugarizer directory for eslint

NikhilM98 commented:
@parteekcoder the `sugarizer` directory should not be inside the `sugarizer-server` project. The `Moving Sugarizer Directory` task does not moves Sugarizer, it makes a copy of the project.

Maybe you can try something like this:
```
...
      - name: Cloning Sugarizer Repository
        run: git clone https://github.com/llaske/sugarizer.git /home/runner/work/sugarizer-server/
      - name: Running Clean Install
        run: npm ci
...
```

parteekcoder commented:
yeah @NikhilM98 thanks for the help , I will look into it

parteekcoder committed:
Update testingWorkflow.yml

parteekcoder committed:
Update .eslintignore

parteekcoder committed:
Update .eslintignore

parteekcoder commented:
done @NikhilM98 great cordination thanks for the help,

passing PR:
https://github.com/parteekcoder/sugarizer-server/actions/runs/4342422084
failing PR:
https://github.com/parteekcoder/sugarizer-server/actions/runs/4335641942/jobs/7584971453

NikhilM98 commented:
@parteekcoder I don't see any point in having 2 steps for cloning and moving the Sugarizer project.

Doesn't something like `git clone https://github.com/llaske/sugarizer.git /home/runner/work/sugarizer-server/` or `git clone https://github.com/llaske/sugarizer.git ../` works?

NikhilM98 commented:
Also, there are too many commits on this PR right now, can you squash your commits?

parteekcoder commented:
> @parteekcoder I don't see any point in having 2 steps for cloning and moving the Sugarizer project.
> 
> Doesn't something like `git clone https://github.com/llaske/sugarizer.git /home/runner/work/sugarizer-server/` or `git clone https://github.com/llaske/sugarizer.git ../` works?

i think that would also work , do you want me to do like that or is it fine

NikhilM98 commented:
> i think that would also work , do you want me to do like that or is it fine

It's be better to avoid unnecessary stages, so it'll be appreciated if you can do that.

parteekcoder committed:
Merge branch 'workflow_CI' of https://github.com/parteekcoder/sugarizer-server into workflow_CI

parteekcoder committed:
fixes eslint issue

Update testingWorkflow.yml

Update testingWorkflow.yml

parteekcoder commented:
> @parteekcoder I don't see any point in having 2 steps for cloning and moving the Sugarizer project.
> 
> Doesn't something like `git clone https://github.com/llaske/sugarizer.git /home/runner/work/sugarizer-server/` or `git clone https://github.com/llaske/sugarizer.git ../` works?

it is not working as git clone only clone repo in empty directory , so have to go with 2 steps @NikhilM98

NikhilM98 commented:
@parteekcoder I just noticed the CI failure. Try this `git clone https://github.com/llaske/sugarizer.git ../sugarizer`.

parteekcoder commented:
> CI failure. Try thi

ok I will try it

parteekcoder commented:
yes @NikhilM98 It is working , I updated the PR

passing test
https://github.com/parteekcoder/sugarizer-server/actions/runs/4343459086/jobs/7585526465

NikhilM98 commented:
Great, can you now squash your changes into a single commit?

parteekcoder commented:
It is not allowing me to sqaush  I squashed as much as I can

NikhilM98 commented:
Maybe consider looking a bit more on the internet about squashing last N commits into a single commit.

Since `parteekcoder:workflow_CI` is a branch on your repository, it's not possible not to be "allowed" to compress multiple commits into one.

parteekcoder commented:
sure @NikhilM98 I will also explore that 

Now this fix continue in #384
